src/models/Model.cc

```diff
@@ -3,6 +3,30 @@
 //
 
 #include "Model.h"
+#include "Substitutor.h"
+#include <cstring>
+
+Model::Model(Logic& logic, Evaluation basicEval, SymbolDefinition symbolDef)
+    : varEval(std::move(basicEval))
+    , symDef(std::move(symbolDef))
+    , logic(logic)
+    , formalArgDefaultPrefix("x")
+{
+    assert(std::all_of(symDef.begin(), symDef.end(),
+                       [&logic](const SymbolDefinition::value_type & entry)
+       {
+           SymRef sr = entry.first;
+           const TemplateFunction & templFun = entry.second;
+           if (not logic.isUF(sr)) { std::cout << "Not UF: " << logic.getSymName(sr) << std::endl; return false; }
+           const Symbol & s = logic.getSym(sr);
+           if (templFun.getName() != logic.getSymName(sr)) { std::cout << "Name doesn't match" << std::endl; return false; }
+           if (s.nargs() != templFun.getArgs().size_()) { std::cout << "Argument number doesn't match" << std::endl; return false; }
+           if (logic.getSortRef(sr) != templFun.getRetSort()) { std::cout << "Return sorts don't match" << std::endl; return false; }
+           for (auto i = 0; i < (int)s.nargs(); i++)
+               if (s[i] != logic.getSortRef(templFun.getArgs()[i])) { std::cout << "argument sort doesn't match" << std::endl; return false; }
```

Comment:
Should these be `std::cerr` instead of `std::cout`?

Reply:
I'll just remove the prints.